src/meta/processors/admin/AdminClient.h

```diff
@@ -198,56 +201,105 @@ class AdminClient {
    * @param jobId
    * @param taskId
    * @param spaceId
-   * @param specificHosts if hosts are empty, will send request to all ative storage hosts
+   * @param host Target host to add task
    * @param taskSpecficParas
    * @param parts
    * @param concurrency
-   * @param statsResult
-   * @return folly::Future<Status>
+   * @return folly::Future<StatusOr<bool>> Return true if succeed, else return an error status
    */
-  virtual folly::Future<Status> addTask(cpp2::AdminCmd cmd,
-                                        int32_t jobId,
-                                        int32_t taskId,
-                                        GraphSpaceID spaceId,
-                                        const std::vector<HostAddr>& specificHosts,
-                                        const std::vector<std::string>& taskSpecficParas,
-                                        std::vector<PartitionID> parts,
-                                        int concurrency,
-                                        cpp2::StatsItem* statsResult = nullptr);
+  virtual folly::Future<StatusOr<bool>> addTask(cpp2::AdminCmd cmd,
+                                                int32_t jobId,
+                                                int32_t taskId,
+                                                GraphSpaceID spaceId,
+                                                const HostAddr& host,
+                                                const std::vector<std::string>& taskSpecficParas,
+                                                std::vector<PartitionID> parts,
+                                                int concurrency);
 
   /**
    * @brief Stop stoarge admin task in given storage host
    *
-   * @param target if target hosts are emtpy, will send request to all active storage hosts
+   * @param host Target storage host to stop job
    * @param jobId
    * @param taskId
-   * @return folly::Future<Status>
+   * @return folly::Future<StatusOr<bool>> Return true if succeed, else return an error status
    */
-  virtual folly::Future<Status> stopTask(const std::vector<HostAddr>& target,
-                                         int32_t jobId,
-                                         int32_t taskId);
+  virtual folly::Future<StatusOr<bool>> stopTask(const HostAddr& host,
```

Comment:
why don't use Status::OK to represent succeed?

Reply:
We need to unify all interfaces, because some of the interfaces need to return extra data, for example, `create snapshot`, which means `Status` is not enough. So some interfaces uses `StatusOr<bool>`